thrifty-schema/src/main/java/com/microsoft/thrifty/schema/Schema.java

```diff
@@ -107,4 +117,73 @@ public EnumType findEnumByType(ThriftType type) {
         }
         throw new NoSuchElementException("No enum type matching " + type.name());
     }
+
+    public Builder toBuilder() {
+        return new Builder(structs, unions, exceptions, enums, constants, typedefs, services);
+    }
+
+    public static final class Builder {
+        private ImmutableList<StructType> structs;
+        private ImmutableList<StructType> unions;
+        private ImmutableList<StructType> exceptions;
+        private ImmutableList<EnumType> enums;
+        private ImmutableList<Constant> constants;
+        private ImmutableList<Typedef> typedefs;
+        private ImmutableList<Service> services;
+
+        Builder(ImmutableList<StructType> structs,
+                ImmutableList<StructType> unions,
+                ImmutableList<StructType> exceptions,
+                ImmutableList<EnumType> enums,
+                ImmutableList<Constant> constants,
+                ImmutableList<Typedef> typedefs,
+                ImmutableList<Service> services) {
+            this.structs = structs;
+            this.unions = unions;
+            this.exceptions = exceptions;
+            this.enums = enums;
+            this.constants = constants;
+            this.typedefs = typedefs;
+            this.services = services;
+        }
+
+        public Builder structs(ImmutableList<StructType> structs) {
```

Comment:
Commenting once here for all of methods on this builder: we need to check the argument for null here, throwing `NullPointerException` if so.